### PR TITLE
Bootstrap admin user and (dev) project

### PR DESCRIPTION
**Pull request by tommydeboer**

### CODE REVIEW

File: armadillo/src/main/java/org/molgenis/armadillo/metadata/UserDetails.java [hunk 1]
```diff
@@ -53,4 +53,8 @@ public static UserDetails create(
   public static UserDetails create(String newEmail) {
     return new AutoValue_UserDetails(newEmail, null, null, null, null, new HashSet<>());
   }
+
```
Comment (mswertz) [inline armadillo/src/main/java/org/molgenis/armadillo/metadata/UserDetails.java:56]: Duplicate?

File: armadillo/src/main/resources/application.yml [hunk 2]
```diff
@@ -7,6 +7,12 @@ rserve:
 datashield:
   # set this to 'true' if you want to accept permissions from oidc provider
   oidc-permission-enabled: false
+
+  # Uncomment this section to bootstrap the application with default values.
+  # bootstrap:
+    # oidc-admin-user: user@yourdomain.org
+    # default-project: your-project
```
Comment (mswertz) [inline armadillo/src/main/resources/application.yml:14]: what is this for?

File: armadillo/src/test/java/org/molgenis/armadillo/controller/AdminControllerTest.java [hunk 3]
```diff
@@ -39,20 +43,32 @@
 @ActiveProfiles("test")
 @Import({AuditEventPublisher.class})
 class AdminControllerTest {
+
   public static final String EXAMPLE_SETTINGS =
```
Comment (mswertz) [inline armadillo/src/test/java/org/molgenis/armadillo/controller/AdminControllerTest.java:47]: is this one of needed than?

**Review by mswertz — APPROVED** [review]
code review looks good. i am scared how this impacts #174 :-)

File: armadillo/src/main/java/org/molgenis/armadillo/metadata/UserDetails.java (hunk 1 above)
Comment (tommydeboer) [reply armadillo/src/main/java/org/molgenis/armadillo/metadata/UserDetails.java:56]: No it's specifically for easily creating an admin. `create` vs `createAdmin`

File: armadillo/src/main/resources/application.yml (hunk 2 above)
Comment (tommydeboer) [reply armadillo/src/main/resources/application.yml:14]: `default-project` can be used during development. In the data folder we have the project `lifecycle` but every time you remove/reset the system folder with the metadata, you have to manually add lifecycle as a project again. Now you can just create your own `application-local.yml` (see readme), configure that project once and be done with it.

File: armadillo/src/test/java/org/molgenis/armadillo/controller/AdminControllerTest.java (hunk 3 above)
Comment (tommydeboer) [reply armadillo/src/test/java/org/molgenis/armadillo/controller/AdminControllerTest.java:47]: Yes it was still used by a test that asserts the output of the "get all metadata" endpoint

**Comment by tommydeboer** [comment]
> code review looks good. i am scared how this impacts https://github.com/molgenis/molgenis-service-armadillo/pull/174 :-)

🤐 
Let me know if you need help :)

File: armadillo/src/main/java/org/molgenis/armadillo/metadata/UserDetails.java (hunk 1 above)
Comment (mswertz) [reply armadillo/src/main/java/org/molgenis/armadillo/metadata/UserDetails.java:56]: ah, I should read better, thx.

File: armadillo/src/main/resources/application.yml (hunk 2 above)
Comment (mswertz) [reply armadillo/src/main/resources/application.yml:14]: that is great.

**Pull request merged**